File: source/bxgenbb_help/src/decay0/Ti48low.cc

#include <sstream>
#include <stdexcept>
#include <cmath>

#include <mygsl/rng.h>

#include <genbb_help/decay0/Ti48low.h>
#include <genbb_help/primary_event.h>
#include <genbb_help/decay0/alpha.h>
#include <genbb_help/decay0/gamma.h>
#include <genbb_help/decay0/electron.h>
#include <genbb_help/decay0/positron.h>
#include <genbb_help/decay0/particle.h>
#include <genbb_help/decay0/pair.h>
#include <genbb_help/decay0/nucltransK.h>
#include <genbb_help/decay0/nucltransKL.h>
#include <genbb_help/decay0/nucltransKLM.h>
#include <genbb_help/decay0/nucltransKLM_Pb.h>
#include <genbb_help/decay0/beta.h>
#include <genbb_help/decay0/beta1.h>
#include <genbb_help/decay0/beta2.h>
#include <genbb_help/decay0/beta_1fu.h>
#include <genbb_help/decay0/PbAtShell.h>

namespace genbb {
  namespace decay0 {

    // Ti48low.f 
    // This file was extracted from the 'decay0' program by V.I. Tretyak
    // Copyright 1995-2011 V.I. Tretyak
    // This program is free software
    // it under the terms of the GNU General Public License as published by
    // the Free Software Foundation
    // your option) any later version.
    // 
    // This program is distributed in the hope that it will be useful, but
    // WITHOUT ANY WARRANTY
    // MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
    // General Public License for more details.
    // 
    // You should have received a copy of the GNU General Public License
    // along with this program
    // Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
    // 

    void Ti48low(mygsl::rng & prng_, genbb::primary_event & event_, int levelkev)
    {
      double t;
      double tdlev;
      double p;
      double tclev;
      double thlev;
      // Subroutine describes the deexcitation process in Ti48 nucleus
      // after 2b-decay of Ca48 to ground and excited 0+ and 2+ levels
      // of Ti48 ("Table of Isotopes", 7th ed., 1978).
      // Call : call Ti48low(levelkev)
      // Input : levelkev - energy of Ti48 level (integer in keV) occupied
      // initially
      // 0+(gs) - 0 keV,
      // 2+(1) - 984 keV,
      // 2+(2) - 2421 keV.
      // Output: // common/genevent/tevst,npfull,npgeant(100),pmoment(3,100),// ptime(100).
      // VIT, 19.07.1993, 22.10.1995.
      tclev=0.;
      if (levelkev == 2421) goto label_2421;
      if (levelkev ==  984) goto label_984;
      if (levelkev ==  0) goto label_10000;
      goto label_20000;
    label_2421  :
      thlev=24.e-15;
      p=100.*prng_();
      if (p <= 8.) goto label_24211;
      goto label_24212;
    label_24211  :
      decay0_nucltransK(prng_, event_, 2.421,0.005,1.5e-5,5.0e-4,tclev,thlev,tdlev);
      return;
    label_24212  :
      decay0_nucltransK(prng_, event_, 1.437,0.005,3.1e-5,1.8e-4,tclev,thlev,tdlev);
      goto label_984;
    label_984  :
      thlev=4.3e-12;
      decay0_nucltransK(prng_, event_, 0.984,0.005,1.2e-4,0.,tclev,thlev,tdlev);
      return;
    label_10000  :
      return;
    label_20000  :
      // print *,'Ti48: wrong level [keV] ',levelkev
      return;
    }
    // end of Ti48low.f




  } // end of namespace decay0 
} // end of namespace genbb 

// end of Ti48low.cc
// Local Variables: --
// mode: c++ --
// End: --

File: source/bxgenbb_help/src/decay0_fortran/Ti48low.f
c Ti48low.f 
c
c This file was extracted from the 'decay0' program by V.I. Tretyak
c
c Copyright 1995-2011 V.I. Tretyak
c
c This program is free software; you can redistribute it and/or modify
c it under the terms of the GNU General Public License as published by
c the Free Software Foundation; either version 3 of the License, or (at
c your option) any later version.
c 
c This program is distributed in the hope that it will be useful, but
c WITHOUT ANY WARRANTY; without even the implied warranty of
c MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
c General Public License for more details.
c 
c You should have received a copy of the GNU General Public License
c along with this program; if not, write to the Free Software
c Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
c 
c
	subroutine Ti48low(levelkeV)
c Subroutine describes the deexcitation process in Ti48 nucleus
c after 2b-decay of Ca48 to ground and excited 0+ and 2+ levels
c of Ti48 ("Table of Isotopes", 7th ed., 1978).
c Call  : call Ti48low(levelkeV)
c Input : levelkeV - energy of Ti48 level (integer in keV) occupied
c                    initially; following levels can be occupied:
c                    0+(gs) -    0 keV,
c                    2+(1)  -  984 keV,
c                    2+(2)  - 2421 keV.
c Output: common/genevent/tevst,npfull,npgeant(100),pmoment(3,100),ptime(100).
c VIT, 19.07.1993, 22.10.1995.
	tclev=0.
	if(levelkev.eq.2421) go to  2421
	if(levelkev.eq. 984) go to   984
	if(levelkev.eq.   0) go to 10000
	                     go to 20000
c-----------------------------------------------------------------------
2421	thlev=24.e-15
	p=100.*rnd1(d)
	if(p.le.8.) go to 24211
	            go to 24212
24211	call nucltransK(2.421,0.005,1.5e-5,5.0e-4,tclev,thlev,tdlev)
	return
24212	call nucltransK(1.437,0.005,3.1e-5,1.8e-4,tclev,thlev,tdlev)
	go to 984
c-----------------------------------------------------------------------
984	thlev=4.3e-12
	call nucltransK(0.984,0.005,1.2e-4,0.,tclev,thlev,tdlev)
	return
c-----------------------------------------------------------------------
10000	return
c-----------------------------------------------------------------------
20000	print *,'Ti48: wrong level [keV] ',levelkev
c-----------------------------------------------------------------------
	return
	end
c
c end of Ti48low.f
c
c Local Variables: --
c mode: fortran --
c End: --
